### PR TITLE
Add architecture documentation

### LINKED ISSUE #119: Write a tech report to introduce the architecture of Processing.R
Now I have invited some friends of me to contribute to Processing.R, so we need a article to illustrate the architecture of Processing.R to help them get started

### PR DESCRIPTION
**Pull request by gaocegege**
Close #119 

Signed-off-by: Ce Gao <ce.gao@outlook.com>

### CODE REVIEW

**Comment by gaocegege** [comment]
PTAL @jeremydouglass @hawkingrei

File: raw-docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+`RLangMode` is the main class for the Processing.R PDE mode. It is accessed through reflection. Processing.R uses Java Remote Method Invocation(RMI), to solve the performance problem. Processing.R mode creates some long-running processes(`SketchServiceRunner`) to handle the requests from Processing Development Environment(PDE).
+
+#### `RLangEditor`
+
+`RLangEditor` is the class that represents the built-in text editor for R mode in the Processing Development Environment(PDE). It extends PDE's `Editor`, builds the mode editor, and initializes all related components, such as a formatter toolbar.
+
+#### `ModeService` and `SketchServiceManager`
+
+`ModeService` is a Remote Method Invocation(RMI) interface, which extends `java.rmi.Remote`. `SketchServiceManager` is the class which implements `ModeService`. `RLangMode` creates a `SketchServiceManager` instance when started. `SketchServiceManager` manages all `SketchServiceRunner`s, and `SketchServiceRunner` runs the real logic of Processing.R sketches.
```
Comment (gaocegege) [inline raw-docs/architecture.md]: s/a/an?

Comment (jeremydouglass) [inline raw-docs/architecture.md]: Not sure I understand. Is this sed for changing to "an Remote Method Invocation" and "an SketchServiceManager" ? Those changes aren't  needed.

"a" comes before closed consonant letters and sounds here like "Remote" and "Sketch", "an" would come before open vowel letters and sounds like "Interface" or "Extends."

Comment (gaocegege) [inline raw-docs/architecture.md]: OK, I got it!

**Comment by gaocegege** [comment]
LGTM 🎉

File: raw-docs/architecture.md (hunk 1 above)
Comment (jeremydouglass) [inline raw-docs/architecture.md]: In spoken English, people would say "a Remote" (AY reemOHT). However, they would say "an R.M.I." (AAN AHR-EHM-IY). Changing "a" to "an" is a way to keep the two vowel sounds (AY AHR) from blending together.

**Pull request merged**